### DISCUSSION TITLE
Updating Column that is an Array

### DISCUSSION THREAD

**Discussion by Kalsyc** [sqlalchemy/sqlalchemy#8799] [Usage Questions] [answered]
Hi guys, new to sqlalchemy and backend with python here.
I'm using postgresql with the orm api in sqlalchemy and i would like to update a column that is an array of uuids:

Essentially, I have a UsersTable and a WorkspacesTable. If I create a workspace, i would need to update the correct row in UsersTable with the newly created workspace_id in the Workspaces column. However, i'm having trouble understanding how to do that. So far I have:

`db.execute(update(UsersTable).where(UsersTable.User_ID == user_id).values(Workspaces=f"{new_workspace_id}"))`

^ This is definitely wrong because what i want to do is to append to the workspaces column so a user can be attached to multiple workspaces. How would I go about creating the query for that?

I have seen a similar post asking for the same help, but they suggest using MutableList? Is there a way with using a normal array column?

**Comment by CaselIT** [comment, marked as answer]
Hi,

How is the workspaces columm defined? If it's a plain array you will need to use the appropriate postgresql function.
https://www.postgresql.org/docs/current/functions-array.html
In this case it's probably `array_append`

**Reply by Kalsyc** [reply]
Yes, you are right. I forgot about that function.

Also found this in case someone needs help too:
https://stackoverflow.com/questions/54127109/updating-a-postgresql-array-field-in-sqlalchemy

**Reply by CaselIT** [reply]
note that you should be able to use `func.array_append` without needing to use `text` as suggested by the stackoverflow answer
